### PR TITLE
initial implementation

### PR DESCRIPTION
**Pull request by MylesBorins**
🎉

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,28 @@
+{
+  "name": "supports-esm",
+  "version": "1.0.0",
```
Comment (MylesBorins) [inline package.json]:
```suggestion
  "version": "0.0.0",
```

Should we start at 0.0.0 so we can bump the major in the original release? (So we can still use `npm version`

File: index.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+'use strict';
+
+const {satisfies} = require('semver');
+
+function supportsESM(baseVersion=process.version, flags=process.execArgv) {
+  const hasExperimentalModulesFlag = flags.includes('--experimental-modules');
+  // unflagged in 14.x
+  return satisfies(baseVersion, '>=14.x') ||
+    // modern implementation with a flag in 13.7 and higher
+    satisfies(baseVersion, '^13.7') && hasExperimentalModulesFlag ||
```
Comment (targos) [inline index.js]: it's unflagged in v13.7 too

**Comment by targos** [comment]
I prefer we remove package-lock.json and add a .npmrc with `package-lock=false`

File: index.js (hunk 2 above)
Comment (MylesBorins) [reply index.js]: fixed

**Comment by MylesBorins** [comment]
removed the package lock.

I'm a big fan of `npm audit` which this removes... but I guess we could enable dependabot for security releases

**Pull request merged**